health-sdk/sdk/src/main/java/net/gini/android/health/sdk/moreinformation/MoreInformationFragment.kt

```diff
@@ -0,0 +1,201 @@
+package net.gini.android.health.sdk.moreinformation
+
+import android.graphics.drawable.Drawable
+import android.os.Bundle
+import android.text.SpannableString
+import android.text.SpannedString
+import android.text.TextPaint
+import android.text.method.LinkMovementMethod
+import android.text.style.TextAppearanceSpan
+import android.text.style.URLSpan
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ExpandableListAdapter
+import android.widget.ExpandableListView
+import androidx.core.text.buildSpannedString
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.coroutines.launch
+import net.gini.android.health.sdk.R
+import net.gini.android.health.sdk.databinding.GhsFragmentPaymentMoreInformationBinding
+import net.gini.android.health.sdk.databinding.GhsPaymentProviderIconHolderBinding
+import net.gini.android.health.sdk.paymentcomponent.PaymentComponent
+import net.gini.android.health.sdk.paymentcomponent.PaymentProviderAppsState
+import net.gini.android.health.sdk.paymentprovider.PaymentProviderApp
+import net.gini.android.health.sdk.util.autoCleared
+import net.gini.android.health.sdk.util.getLayoutInflaterWithGiniHealthTheme
+
+
+/**
+ * Created by dani on 26/02/2024.
+ */
+
+class MoreInformationFragment private constructor(val paymentComponent: PaymentComponent?) :
```

Comment:
[Suggestion] Please create a ViewModel subclass and use it to avoid crashing when the fragment needs to be recreated. You can see the `BankSelectionBottomSheet` on how I did it. Or if you have a different solution in mind let's discuss it and see how to best solve it.

Here is the exception I got:
```
Process: net.gini.android.health.sdk.exampleapp, PID: 27272
java.lang.RuntimeException: Unable to start activity ComponentInfo{net.gini.android.health.sdk.exampleapp/net.gini.android.health.sdk.exampleapp.invoices.ui.InvoicesActivity}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment net.gini.android.health.sdk.moreinformation.MoreInformationFragment: could not find Fragment constructor
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3676)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3813)
	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5791)
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5682)
	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:71)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2308)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7898)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment net.gini.android.health.sdk.moreinformation.MoreInformationFragment: could not find Fragment constructor
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:687)
	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
	at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:525)
	at androidx.fragment.app.FragmentState.instantiate(FragmentState.java:84)
	at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:91)
	at androidx.fragment.app.FragmentManager.restoreSaveStateInternal(FragmentManager.java:2562)
	at androidx.fragment.app.FragmentManager.attachController(FragmentManager.java:2722)
	at androidx.fragment.app.FragmentController.attachHost(FragmentController.java:117)
	at androidx.fragment.app.FragmentActivity.lambda$init$3$androidx-fragment-app-FragmentActivity(FragmentActivity.java:139)
	at androidx.fragment.app.FragmentActivity$$ExternalSyntheticLambda3.onContextAvailable(Unknown Source:2)
	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.kt:84)
	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:358)
	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:216)
	at net.gini.android.health.sdk.exampleapp.invoices.ui.InvoicesActivity.onCreate(InvoicesActivity.kt:42)
	at android.app.Activity.performCreate(Activity.java:8290)
	at android.app.Activity.performCreate(Activity.java:8269)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1384)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3657)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3813) 
	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5791) 
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5682) 
	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:71) 
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2308) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:201) 
	at android.os.Looper.loop(Looper.java:288) 
	at android.app.ActivityThread.main(ActivityThread.java:7898) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
Caused by: java.lang.NoSuchMethodException: net.gini.android.health.sdk.moreinformation.MoreInformationFragment.<init> []
	at java.lang.Class.getConstructor0(Class.java:2363)
	at java.lang.Class.getConstructor(Class.java:1759)
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:672)
	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57) 
	at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:525) 
	at androidx.fragment.app.FragmentState.instantiate(FragmentState.java:84) 
	at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:91) 
	at androidx.fragment.app.FragmentManager.restoreSaveStateInternal(FragmentManager.java:2562) 
	at androidx.fragment.app.FragmentManager.attachController(FragmentManager.java:2722) 
	at androidx.fragment.app.FragmentController.attachHost(FragmentController.java:117) 
	at androidx.fragment.app.FragmentActivity.lambda$init$3$androidx-fragment-app-FragmentActivity(FragmentActivity.java:139) 
	at androidx.fragment.app.FragmentActivity$$ExternalSyntheticLambda3.onContextAvailable(Unknown Source:2) 
	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.kt:84) 
	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:358) 
	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:216) 
	at net.gini.android.health.sdk.exampleapp.invoices.ui.InvoicesActivity.onCreate(InvoicesActivity.kt:42) 
	at android.app.Activity.performCreate(Activity.java:8290) 
	at android.app.Activity.performCreate(Activity.java:8269) 
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1384) 
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3657) 
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3813) 
	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5791) 
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5682) 
	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:71) 
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2308) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:201) 
	at android.os.Looper.loop(Looper.java:288) 
	at android.app.ActivityThread.main(ActivityThread.java:7898) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
```

Reply:
Thank you for pointing this out - changed to retrieving the `PaymentAppProvidersList` in a new `MoreInformationViewModel` instead of handling it in the activity directly. 

Also check and set title in `InvoicesActivity` when it is recreated, if `MoreInformationFragment` is the last one in the stack.